content/knowledge/scalinghistory.md

```diff
@@ -0,0 +1,144 @@
+In PoS, explicit agreement is required before a block becomes valid.
+If a sufficient number of validators do not sign the block then it will be rejected even if all consensus rule is followed by the block.
+Only after the necessary number of participants to the consensus algorithm explicitly vote for the block by signing it, will it become valid.
+This means that a lot of communication is needed.
+In general, voting on the next correct block requires multiple rounds of communication (preparation round, confirmation round and commit round), and the more validators there are, the more overhead there is for everyone to get their messages to the leader of a round.
+Due to this high number of communication scalability is hindered.
+
+Also the higher the adaptation, the higher the total value represented by the native cryptocurrency and therefore higher the value that needs to be locked up as staking by the nodes.
+This means the native currency's full potential is not utilised as once they are locked, they can no longer be used for any other purposes unless they are unlocked after the locking period expires.
+Network’s security is proportional to the value of stake committed.
+
+## Sharding and layer 2 solutions
+
+For solving this scaling issue, 'sharding' is introduced (e.g. in the case of [Ethereum 2.0](https://www.youtube.com/watch?v=ctzGr58_jeI&t=657s&ab_channel=Finematics)).
+Sharding is a way of spreading the computing and storage workload from a blockchain network, so that each node no longer has to process the entire network's transactional load.
+Each node only needs to maintain info corresponding to its specific shard or partition.
+For example, addresses starting with “0x00” will be part of shard 1 and addresses starting with “0x01” will be part of shard 2 etc.
+There is a mechanism in place for inter-shard communication so that the whole blockchain is still viewable from any node.
+All the shards will be connected to the “Beacon chain” which will have all the history with it and acts as an orchestrator of the whole network.
+The consensus between all the shards will be maintained by a "beacon chain".
+However, sharding has its own challenges.
+Corrupting nodes in a given shard may lead to permanent loss of data.
+One way to tackle this issue is by randomly assigning a node to a shard and randomly reassigning the node to another shard.
+Sharding has another problem that the composability aspect of smart contract compatible blockchain will no longer exist.
+For example, if a Decentralized Finance (DeFi) platform built on top of Ethereum relies on relies on or combines few other projects, then in a sharded system of Ethereum2.0, ideally they should be on the same shard which may not always be the case.
+If they are not on the same shard then they all needs to be ok with asynchronous updates which is [not ideal](https://www.coindesk.com/tech/2020/10/13/will-a-sharded-ethereum-be-flexible-enough-for-decentralized-finance/).
```

Comment:
Phrasing it like "they all need to be ok with asynchronous updates" is a bit weird, and informal. I lack a little context here, but perhaps something like "they all need to support asynchronous updates" might fit better. WDYT?